File: src/main/java/com/circleescape/server/model/game/GameEntity.java
package com.circleescape.server.model.game;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "GAMES")
public class GameEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Embedded
    private GameParameters gameParameters;

    @Column
    private String authKey;

    @ElementCollection
    private List<Collocation> collocations = new ArrayList<>();

}

File: src/main/kotlin/com/circleescape/server/model/game/persistence/GameEntity.kt
package com.circleescape.server.model.game.persistence

import javax.persistence.*

@Entity
@Table(name = "GAMES")
class GameEntity(
    var authKey: String, //TODO use it
    @Embedded var gameParameters: GameParameters,
    @ElementCollection var collocations: List<Collocation> = ArrayList(),
    var status: String,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long? = null
)